### ISSUE TITLE
Just... can't get it to work.

### ISSUE THREAD

**Issue by mikeevmm** [vshaxe/vshaxe#117]
Much like the title says; I simply can't get the VSCode Haxe extension to work, and can't find the problem reproduced anywhere.

The Haxe server initializes normally, with good output. Here it is after a restart:

```
Haxe language server started
Initializing completion cache...
Done.
```

Syntax highlighting is working, but no autocompletion or intellisense in general.

When I press `ctrl+space` I just get a `loading` floating box. Same with `Show all references`.

I suspect this might have to do with both my Haxe and VSCode installation residing in E:/ (as opposed to C:/), but it's a wild guess.

I'm using the standard `./vscode` and other files output from `>Haxe: Initialize VSCode project`, with the exception of `build.hxml`, which is as follows:

```
-cp src
-D analyzer-optimize
-main Main
-cpp keratin
```

I've tried restarting the Haxe server many times, as well as reinstalling the extension, and even the haxe installation. Building a project works fine.

It's like intellisense is just.. hanging, internally.

**Comment by nadako** [comment]
This is very weird... The hxml args look fine. Could you try adding the following to `.vscode/settings.json`?

```json
    "haxe.displayServer": {
        "arguments": [
            "-v"
        ]
    }
```

It should start haxe completion server in verbose mode so we can see more debug information.

**Comment by mikeevmm** [comment]
@nadako 

```
Haxe language server started
Initializing completion cache...
Processing Arguments [build.hxml,--no-output]
Defines analyzer-optimize=1,analyzer_optimize=1,cpp=1,dce=std,haxe3=1,haxe_ver=3.402,hxcpp_api_level=331,source-header=Generated by Haxe 3.4.2 (git build master @ 890f8c7),sys=1,true=1
Using signature a0fb0def2cdab5f5c9b8a8974589e9b6
Display position: ?: -1--1
Cached 21 modules
Stats = 27 files, 28 classes, 69 methods, 0 macros
Time spent : 0.018s
Done.
Processing Arguments [--cwd,e:\Documents\Investigação\haxe,build.hxml,-D,display-details,--no-output,--display,e:/Documents/Investigação/haxe/src/Main.hx@0@diagnostics]
Stats = 0 files, 0 classes, 0 methods, 0 macros
Time spent : 0.000s
```

Can't see anything wrong.

**Comment by mikeevmm** [comment]
(In case you're wondering if those `ç`s and `ã`s in the path affect the extension; they don't, I can't get working behaviour anywhere.)

**Comment by nadako** [comment]
and there are no more `Processing Arguments` log entries when tryigng to invoke completion?

**Comment by mikeevmm** [comment]
Doesn't seem like it.

**Comment by Gama11** [comment]
The shortcut for invoking completion is <kbd>Ctrl</kbd>+<kbd>Space</kbd> actually, not <kbd>Ctrl</kbd>+<kbd>Enter</kbd>. Not sure if typing that sequence in particular would trigger a completion request, hovering over something should though.

**Comment by mikeevmm** [comment]
Sorry, I meant `Ctrl`+`Space`.

**Comment by Gama11** [comment]
Ok. Anyway, that means you're not getting any output when hovering over expressions either? The hover hint UI just shows "Loading..." then?

**Comment by mikeevmm** [comment]
Yes.

**Comment by Gama11** [comment]
Did you ever try this in an empty project, or did you use the "init project" command on an already existing project to add the .vscode directory?

Would you mind cloning this repo, to make sure it's not an issue with your vscode config files?

https://github.com/Gama11/HaxeRepro

You should get completion features like hover hints out of the box with that:

![](http://i.imgur.com/QEy2sqy.png)

**Comment by mikeevmm** [comment]
I can get completion features with the clone repo. This is very strange, as the projects I'd been working with were fresh, pure Haxe cmd generated project folders.

However, I did update the Haxe extension in the meantime, so I'll go test again with my other projects.

**Comment by mikeevmm** [comment]
The issue persists with older projects.

**Comment by Gama11** [comment]
Can you share one of those projects, to see what may be wrong with the configuration?

**Comment by mikeevmm** [comment]
Of course:

https://github.com/mikeevmm/HaxeVSCodeIssue

Project created with `Init Project` suffers from the same.

**Comment by Gama11** [comment]
Completion works just fine for me in a freshly cloned version of that repo..

![](http://i.imgur.com/SMGF0so.png)

**Comment by mikeevmm** [comment]
😧

Maybe I'm missing something though; my VSCode instalation looks diferent:

![image](https://cloud.githubusercontent.com/assets/7550678/26148985/669e45ce-3af0-11e7-95af-d4716dbf960f.png)

**Comment by Gama11** [comment]
No, it's just that I have [vscode-icons](https://marketplace.visualstudio.com/items?itemName=robertohuertasm.vscode-icons) installed.

**Comment by mikeevmm** [comment]
Ah..

I have no idea then.

**Comment by mikeevmm** [comment]
I ran some tests; It might have to do with files residing on `E:/`, rather than `C:/`.

**Comment by mikeevmm** [comment]
Confirmed: @Gama11's project suffers from the issue when placed on `E:/`, and my own project has autocompletion when placed somewhere on `C:/`.

**Comment by AG-w** [comment]
Can confirmed
I have a project that autocompletion will stuck at loading

but if I copy my work from E:/ to C:/, it works perfectly

**Comment by nadako** [comment]
wow, that's super weird. is it really only about the drive letter? maybe some weird folder name with spaces and unicode characters?

**Comment by Jarrio** [comment]
Is this only an issue with the `E:` drive? I have my VSCode installation on a `P:` drive extensions are installed on the `C:` drive and my work is on a `W:` drive and parsing works fine for me

**Comment by AG-w** [comment]
I got it working  in E:/ this time.
I found that If path contain non-English character, auto completion will stop work

**Comment by mikeevmm** [comment]
That's strange... I thought I'd tested it with no foreign chars.. (as [referenced ](https://github.com/vshaxe/vshaxe/issues/117#issuecomment-298119287)).

Will re-test ASAP.

**Comment by mikeevmm** [comment]
I must've missed something: I can indeed get it working if I move to a folder on `E:/` with no non-english chars.

**Comment by nadako** [comment]
I can confirm it's broken with cyrillic characters as well. Though it seems to be an issue with Haxe compiler, not with vshaxe. Will test further.

**Comment by nadako** [comment]
can't reproduce it with command-line `--wait`/`--connect` though, so might be a language server issue after all... testing more.

**Comment by Gama11** [comment]
May be the same issue as #68, where workspace symbols don't work with an apostrophe in the path?

**Comment by nadako** [comment]
This looks like a wrong packet length sent in `--wait stdio` mode, so probably a haxe issue after all.

**Comment by nadako** [comment]
So, this is indeed a haxe issue. Two, in fact. One is that it printed to stderr in server mode, while it's reserved for the services API interaction in `--wait stdio` mode (which I originally thought it's a packet length problem). Fixed in https://github.com/HaxeFoundation/haxe/commit/5843874b2243bb718dccdc03d945a5f921fcf487 (and cherry-picked into 3.4 bugfix branch).

The other issue is something I'm not sure how to deal with yet. VSCode adds `--cwd <workspaceDir>` to the compiler arguments and that's encoded in UTF-8, which is then passed to the `chdir` system call within the compiler. But apparently non-latin letters won't work with UTF-8 on windows and `chdir` expects a string in some filesystem encoding, but I'm really not sure how does that stuff works, so this will require more research.

**Comment by Gama11** [comment]
@nadako Could we simply do `Sys.setCwd(workspacePath);` as a workaround instead of sending `--cwd`?

**Comment by kLabz** [comment]
> VSCode adds `--cwd <workspaceDir>` to the compiler arguments

Wait, what? Is there a way to disable this? This breaks projects relying on `--cwd`.
